kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/TopicsResource.java

```diff
@@ -137,24 +138,38 @@ public void createTopic(
     String topicName = request.getTopicName();
     Optional<Integer> partitionsCount = request.getPartitionsCount();
     Optional<Short> replicationFactor = request.getReplicationFactor();
+    Map<Integer, List<Integer>> replicasAssignments = request.getReplicasAssignments();
     Map<String, Optional<String>> configs =
         request.getConfigs()
             .stream()
             .collect(Collectors.toMap(ConfigEntry::getName, ConfigEntry::getValue));
 
+    // We have no way of knowing the default replication factor in the Kafka broker. Also in case
+    // of explicitly specified partition-to-replicas assignments, all partitions should have the
+    // same number of replicas.
+    final short assumedReplicationFactor = replicationFactor.orElse(
```

Comment:
We don't use `final` for local variables.

Reply:
Fixed.